pkg/ipcache/cidr.go

```diff
@@ -70,6 +70,13 @@ func (ipc *IPCache) AllocateCIDRs(
 	ipc.Unlock()
 	ipc.metadata.RUnlock()
 
+	// Insert any newly allocated identities in to the policy engine
+	addedIdentities := make(map[identity.NumericIdentity]labels.LabelArray, len(newlyAllocatedIdentities))
+	for _, id := range newlyAllocatedIdentities {
+		addedIdentities[id.ID] = id.LabelArray
+	}
+	ipc.UpdatePolicyMaps(context.TODO(), addedIdentities, nil)
+
```

Comment:
During my initial look, I was a little concerned with the potential for race conditions regarding the identity allocation and updating the policymaps. However, having reviewed commit 40e13ea2a5a944a45761fc433c4c971536957f4b I don't think there is a consequential race condition from updating the policy engine here.

Reply:
Agreed; in `injectLabels()` we call `UpdatePolicyMaps()` holding no locks as well. Since identities themselves are immutable, I think this is OK.

All of this is somewhat academic, since there's only one caller of this method left, and it's single-threaded.